src/test/java/seedu/address/testutil/EditPropertyDescriptorBuilder.java

```diff
@@ -31,41 +36,83 @@ public EditPropertyDescriptorBuilder(Property property) {
         descriptor.setPrice(property.getPrice());
         descriptor.setAddress(property.getAddress());
         descriptor.setDescription(property.getDescription());
+        descriptor.setOwner(property.getOwner());
+        descriptor.setOwnerName(property.getOwner().getName());
+        descriptor.setOwnerPhone(property.getOwner().getPhone());
     }
 
     /**
      * Sets the {@code PropertyName} of the {@code EditPropertyDescriptor} that we are building.
      */
-    public EditPropertyDescriptorBuilder withName(PropertyName name) {
-        descriptor.setName(name);
+    public EditPropertyDescriptorBuilder withName(String name) {
+        descriptor.setName(new PropertyName(name));
         return this;
     }
 
     /**
      * Sets the {@code Price} of the {@code EditPropertyDescriptor} that we are building.
      */
-    public EditPropertyDescriptorBuilder withPrice(Price price) {
-        descriptor.setPrice(price);
+    public EditPropertyDescriptorBuilder withPrice(String price) {
+        descriptor.setPrice(new Price(price));
         return this;
     }
 
     /**
      * Sets the {@code Address} of the {@code EditPropertyDescriptor} that we are building.
      */
-    public EditPropertyDescriptorBuilder withAddress(Address address) {
-        descriptor.setAddress(address);
+    public EditPropertyDescriptorBuilder withAddress(String address) {
+        descriptor.setAddress(new Address(address));
         return this;
     }
 
     /**
      * Sets the {@code Description} of the {@code EditPropertyDescriptor} that we are building.
      */
-    public EditPropertyDescriptorBuilder withDescription(Description description) {
-        descriptor.setDescription(description);
+    public EditPropertyDescriptorBuilder withDescription(String description) {
+        descriptor.setDescription(new Description(description));
         return this;
     }
 
-    public EditPropertyDescriptor build() {
+    /**
+     * Sets the {@code Characteristics} of the {@code EditPropertyDescriptor} that we are building.
+     */
+    public EditPropertyDescriptorBuilder withCharacteristics(String characteristics) {
+        descriptor.setCharacteristics(new Characteristics(characteristics));
+        return this;
+    }
+
+    /**
+     * Sets the {@code Owner}, {@code Name} and {@code Phone} of the {@code EditPropertyDescriptor}
+     * that we are building.
+     */
+    public EditPropertyDescriptorBuilder withOwner(String ownerName, String ownerPhone) {
+        Name newOwnerName = new Name(ownerName);
+        Phone newOwnerPhone = new Phone(ownerPhone);
+        descriptor.setOwner(new Owner(newOwnerName, newOwnerPhone));
+        descriptor.setOwnerName(newOwnerName);
+        descriptor.setOwnerPhone(newOwnerPhone);
```

Comment:
We call `setOwner` here, but not in the `withOwnerName` and `withOwnerPhone` methods, even though the `Owner` in the descriptor also changed if we just edit ownerName and ownerPhone.

Maybe we could actually remove the `Owner` field in the descriptor? It seems sufficient to just track the name and phone, since both make up the `Owner`.

Reply:
I thought about this as well, but I figured that it might be good to retain Owner for now just in case we might need it in the future. I also checked the dependencies and nothing critical seems to depend on it, but it's a nice-to-have feature which works as of now.